modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -86,7 +93,8 @@ TensorMechanicsAction::TensorMechanicsAction(const InputParameters & params)
     _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
     _cylindrical_axis_point1_valid(params.isParamSetByUser("cylindrical_axis_point1")),
     _cylindrical_axis_point2_valid(params.isParamSetByUser("cylindrical_axis_point2")),
-    _direction_valid(params.isParamSetByUser("direction"))
+    _direction_valid(params.isParamSetByUser("direction")),
+    _eigenstrain_names_valid(params.isParamSetByUser("eigenstrain_names"))
```

Comment:
not a good name for that parameter. also I suggest not doing this test at all. Pre populate the list of eigenstrain names with the user supplied values, then add on teh autodetected names!